lib/Doctrine/Common/Proxy/ProxyGenerator.php

```diff
@@ -943,6 +973,38 @@ private function getParameterType(\ReflectionParameter $parameter)
         return $this->formatType($parameter->getType(), $parameter->getDeclaringFunction(), $parameter);
     }
 
+	/**
+	 * @param \ReflectionMethod $method
+	 * @param string $parameterName
+	 *
+	 * @return string|null
+	 */
+    private function getParameterTypeByMethod(\ReflectionMethod $method, string $parameterName)
```

Comment:
As mentioned above, we should use positional parameter indexes instead: that removes ambiguity between proxy naming and original class naming